src/arp.rs

```diff
@@ -1,65 +1,189 @@
-use std::ptr::copy_nonoverlapping;
+use crate::pv::PvPacket;
+use crate::utils::*;
+use pnet::datalink::MacAddr;
 
-use crate::pv::{PvPacket};
-use pnet::datalink::{MacAddr};
+pub fn get_hw_type(packet: &PvPacket) -> Result<u16, &str> {
```

Comment:
Let's think about `Err(&str)` return type.

It's good to notice what or where problems happen with this code for human(customers or developers), unfortunately it makes difficult to handle errors.
For example, if `get_hw_type()` returns `Err(u32)` instead of `Err(&str)`, some function that takes `Err(u32)` can handle the error easily because numbers are easier and simpler to be compared `u32` than `&str`.

which one does seem easy to compare values?
```
let err_u32 = Err(1);
let err_str = Err("Error 1");

if err_u32.unwrap() == 1 {
    error_handling_1();
} else if error_u32.unwrap() == 2 {
    error_handling_2();
} else if ... {
    ...
}

if err_str().unwrap().contains("error 1") {
    error_handling_1();
} else if err_str().unwrap().contains("error 2") {
   error_handling_2();
} else if ... {
   ...
}
```